posts/2019/04/uarray-intro.ipynb

```diff
@@ -0,0 +1,473 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# `uarray`: A Generic Override Framework for Methods\n",
+    "\n",
+    "`uarray` is an override framework for methods in Python. In the scientific Python ecosystem, and in other similar places, there has been one recurring problem: That similar tools to do a job have existed, but don't conform to a single, well-defined API. `uarray` tries to solve this problem in general, but also for the scientific Python ecosystem in particular, by defining APIs independent of their implementations.\n",
+    "\n",
+    "## The Scientific Python Ecosystem: How Fragmentation Emerged\n",
+    "When SciPy was created, and Numeric and NumArray unified into NumPy, it jump-started Python's data science community. The ecosystem grew quickly: Academics started moving from MATLAB to SciPy, and the Scikits that popped up made the transition all the more smooth.\n",
+    "\n",
+    "However, the computing community also shifted during that time: GPUs and distributed computing emerged. There were old ideas that couldn't really be used with NumPy's API, such as sparse arrays. To solve these problems, various libraries emerged:\n",
+    "\n",
+    "- Dask, for distributed NumPy\n",
+    "- CuPy, for NumPy on Nvidia-branded GPUs.\n",
+    "- PyData/Sparse, a project started to make sparse arrays conform to the NumPy API\n",
+    "\n",
+    "There were yet other libraries that emerged: PyTorch, which mimics NumPy to a certain degree, and TensorFlow, which defines its own API.\n",
```

Comment:
Also mentioning MXNet, which has an `ndarray` class that closely mimics NumPy would be good. And then at the end something like "and a number of other numpy-like packages".